frameworks/Swift/perfect/Sources/Perfect-MySQL/main.swift

```diff
@@ -0,0 +1,336 @@
+import PerfectHTTP
+import PerfectHTTPServer
+import PerfectLib
+import PerfectMySQL
+import Foundation
+
+let tfbHost = "tfb-database"
+let database = "hello_world"
+let username = "benchmarkdbuser"
+let password = "benchmarkdbpass"
+
+let mysql = MySQL()
+let connected = mysql.connect(host: tfbHost, user: username, password: password)
+let _ = mysql.selectDatabase(named: database)
+
+class LinearCongruntialGenerator {
+ 
+    var state = 0 //seed of 0 by default
+    let a, c, m, shift: Int
+ 
+    //we will use microsoft random by default
+    init() {
+        self.a = 214013
+        self.c = 2531011
+        self.m = Int(pow(2.0, 31.0)) //2^31 or 2147483648
+        self.shift = 16
+    }
+ 
+    init(a: Int, c: Int, m: Int, shift: Int) {
+        self.a = a
+        self.c = c
+        self.m = m //2^31 or 2147483648
+        self.shift = shift
+    }
+ 
+    func seed(seed: Int) -> Void {
+        state = seed;
+    }
+ 
+    func random() -> Int {
+        state = (a * state + c) % m
+        return state >> shift
+    }
+}
+
+let numGenerator = LinearCongruntialGenerator()
+
+func fetchFromFortune() -> [[String: String]] {
+
+    var arrOfFortunes = [[String: String]]()
+    
+    let querySuccess = mysql.query(statement: "SELECT id, message FROM fortune")
+
+    guard querySuccess else {
+
+        let errorObject = ["id": "Failed to execute query"]
+        arrOfFortunes.append(errorObject)
+        
+        return arrOfFortunes
+    }
+ 
+    let results = mysql.storeResults()!
+
+    results.forEachRow { row in
+
+        if let id = row[0], let message = row[1] {
+            
+            let resObj = ["id": String(describing: id), "message": message]
+            arrOfFortunes.append(resObj)
+        } else {
+            print("not correct values returned: ", row)
+        }
+    }
+
+    return arrOfFortunes
+}
+
+func fetchFromWorld(id: String?) -> [String:Any] {
+
+    var returnObj = [String: Any]()
+    var errorObject = [String: Any]()
+    var rand:Int = 0
+
+    if id == nil {
+        rand = numGenerator.random() % 10000
+    } else {
+        rand = Int(id!)!
+    }
+
+    let querySuccess = mysql.query(statement: "SELECT id, randomNumber FROM World WHERE id = \(rand)")
+
+    guard querySuccess else {
+
+        errorObject["id"] = "Failed to execute query"
+
+        return errorObject
+    }
+ 
+    let results = mysql.storeResults()!
+
+    results.forEachRow { row in
+
+        if let id = row[0], let randomNumber = row[1] {
+
+            returnObj["id"] = id
+            returnObj["randomNumber"] = randomNumber
+        } else {
+
+            returnObj["id"] = "No return value"
+            returnObj["randomNumber"] = "what happened?"
+        }
+    }
+
+    return returnObj
+}
+
+func updateOneFromWorld() -> [String: Any] {
+
+    var returnObj = [String: Any]()
+    var errorObject = [String: Any]()
+
+    let rand = numGenerator.random() % 10000
+    let rand2 = numGenerator.random() % 10000
```

Comment:
This is actually 0 - 9999 inclusive. Will need to change all of these to:

`numGenerator.random() % 10000 + 1`

Reply:
Done. Thanks!